src/document.h

```diff
@@ -70,6 +70,9 @@ typedef struct GeanyFilePrefs
 GeanyFilePrefs;
 
 
+#define GEANY_TYPE_DOCUMENT (document_get_type())
+GType document_get_type (void);
```

Comment:
Well, the functions are exported, even if just for the gir. So putting them under GEANY_PRIVATE just isn't right.

Besides, I do generate a (mostly) working .vapi from the .gir. Vala plugins written against that use the C header files so they need to be in the public headers (peasy is partly written in vala).